src/grift/syntax-to-grift0.rkt

```diff
@@ -55,8 +56,11 @@ whatsoever identifiers maintain lexical scope according to symbolic equality.
   (match stx*
     [(list s* ...)
      (define/match (observe/ann e)
-       [((Ann (Define _ _ _ _) s)) e]
-       [((Ann _ s)) (Ann (Observe e #f) s)]
+       ;; Replacing Ann with Ann2 is a hack that prevent
+       ;; the contract generator from getting different Ann
+       ;; types confused.
+       [((Ann (and d (Define _ _ _ _)) s)) (Ann2 d s)]
+       [((Ann _ s)) (Ann2 (Observe e #f) s)]
```

Comment:
I don't think so. There is a bug in contract generation for typed racket. The workaround that I found was to declare two separate instances for Ann, because the contract were being generated where one instantiation of Ann would clobber the other.

Reply:
We should be able to remove this extra form after porting away from typed-racket.